### PR TITLE
Custom extrinsic & Dynamic remote fee

### PR DESCRIPTION
**Pull request by yrong**
Resolves: https://linear.app/snowfork/issue/SNO-1084#comment-0d5970ee

- Add a helper pallet specific for xcm operations(transact, transfer) from Substrate to Ethereum through snowbridge.
- Integrate the helper pallet into runtime of asset hub. 
- Remove the `XcmExportFeeToSibling` on BridgeHub which is unnecessary.
- Revamp integration test to use the new extrinsic `transfer_to_ethereum` for transfer from AH to Ethereum.

### CODE REVIEW

**Comment by yrong** [comment]
In https://github.com/Snowfork/polkadot-sdk/pull/158/commits/0fe04840599aa98798bf410232a35717cbc74886#diff-5ded4b43c234d6f2d43aff2b2b63adfde7a5ba2dd9fc591cd818335e20b6303b I tested with the new extrinsic `transfer_assets_using_type_and_then` and yeah the `send_weth_asset_from_asset_hub_to_ethereum` test case can pass, but does not work the way as we want.

The main issue is the limitation mentioned [here](https://github.com/paritytech/polkadot-sdk/issues/3847) it can't take fees separately.

Say we have 2 assets [(weth_location,1),(weth_location,1)], with the fee index as 0, the intention is to use the first asset as remote fee and send the second to beneficiary.  But there is some hidden logic [here](https://github.com/Snowfork/polkadot-sdk/blob/f7054e73a0069bd59228a79b23c2cb21df6c4e71/polkadot/xcm/src/v4/asset.rs#L565) the 2 assets will be merged into a single one (weth_location,2) and prevent us for taking fees separately.

The extrinsic `transfer_assets_using_type_and_then` won't help here and that's another reason why we add the custom call for that.

**Comment by vgeddes** [comment]
I may be wrong, but my high-level understanding from reading the documentation for `transfer_assets_using_type_and_then` is that fees asset and the asset being transferred can be the same.

Or in other words, the fee amount can be deducted from the asset specified by the user.

In any case, in the new fee design I'm working on, users will most likely pay fees in DOT instead of WETH. The existence of a DEX on AssetHub does make fee currencies other than DOT a bit unnecessary.

**Comment by yrong** [comment]
> users will most likely pay fees in DOT instead of WETH

Yeah, with the [AssetExchanger](https://github.com/paritytech/polkadot-sdk/blob/b0e1c60f5fa944a548bb7e26567ad2b5906aa5d9/cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs#L435) config it's possible.

FYI there is one PR for that in https://github.com/paritytech/polkadot-sdk/pull/4375 I'm reviewing.

**Comment by yrong** [comment]
Close in favor of https://github.com/Snowfork/polkadot-sdk/pull/160

File: bridges/snowbridge/pallets/xcm-helper/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(Weight::from_parts(100_000_000, 0))]
+		pub fn transfer_to_ethereum(
+			origin: OriginFor<T>,
+			beneficiary: H160,
+			asset: Box<VersionedAsset>,
+			fee: Box<VersionedAsset>,
```
Comment (yrong) [inline bridges/snowbridge/pallets/xcm-helper/src/lib.rs:144]: Basically just mirror the `reserve_transfer_assets` extrinsic from `pallet_xcm` with an explicit fee specified here.

Comment (vgeddes) [reply bridges/snowbridge/pallets/xcm-helper/src/lib.rs:144]: Nice 🚀

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs [hunk 2]
```diff
@@ -215,21 +236,14 @@ impl xcm_executor::Config for XcmConfig {
...
-				crate::EthereumOutboundQueue,
-			>,
 			SendXcmFeeToAccount<Self::AssetTransactor, TreasuryAccount>,
 		),
 	>;
 	type MessageExporter = (
 		crate::bridge_to_westend_config::ToBridgeHubWestendHaulBlobExporter,
 		crate::bridge_to_bulletin_config::ToRococoBulletinHaulBlobExporter,
 		crate::bridge_to_ethereum_config::SnowbridgeExporter,
+		crate::bridge_to_ethereum_config::SnowbridgeExporterV2,
```
Comment (yrong) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs:246]: Add SnowbridgeExporterV2 side by side with SnowbridgeExporter so to support two versions at the same time.

**Pull request closed without merging**